testing/integration/upgrade_test.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/elastic/elastic-agent/version"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type UpgradeElasticAgent struct {
+	suite.Suite
+
+	requirementsInfo  *define.Info
+	agentStartVersion string
+	agentEndVersion   string
+	agentFixture      *atesting.Fixture
+}
+
+func newUpgradeElasticAgentTestSuite(info *define.Info, startVersion, endVersion string) *UpgradeElasticAgent {
+	return &UpgradeElasticAgent{
+		requirementsInfo:  info,
+		agentStartVersion: startVersion,
+		agentEndVersion:   endVersion,
+	}
+}
+
+// Before suite
+func (s *UpgradeElasticAgent) SetupSuite() {
+	agentFixture, err := atesting.NewFixture(
+		s.T(),
+		s.agentStartVersion,
+		atesting.WithFetcher(atesting.ArtifactFetcher()),
+	)
+	require.NoError(s.T(), err)
+	s.agentFixture = agentFixture
+}
+
+func (s *UpgradeElasticAgent) TestUpgradeFleetManagedElasticAgent() {
+	policyUUID := uuid.New().String()
+
+	createPolicyReq := kibana.CreatePolicyRequest{
+		Name:        "test-policy-" + policyUUID,
+		Namespace:   "default",
+		Description: "Test policy " + policyUUID,
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+	policy, err := s.requirementsInfo.KibanaClient.CreatePolicy(createPolicyReq)
+	require.NoError(s.T(), err)
+
+	createEnrollmentApiKeyReq := kibana.CreateEnrollmentAPIKeyRequest{
+		PolicyID: policy.ID,
+	}
+	enrollmentToken, err := s.requirementsInfo.KibanaClient.CreateEnrollmentAPIKey(createEnrollmentApiKeyReq)
+	require.NoError(s.T(), err)
+
+	// TODO: figure out how to get Fleet Server URL
+	fleetServerURL := ""
+	output, err := tools.EnrollElasticAgent(fleetServerURL, enrollmentToken.APIKey, s.agentFixture)
+	if err != nil {
+		s.T().Log(string(output))
+	}
+	require.NoError(s.T(), err)
+
+	require.Eventually(s.T(), agentStatus("online", *s), 2*time.Minute, 10*time.Second, "Agent status is not online")
+
+	err = tools.UpgradeAgent(s.requirementsInfo.KibanaClient, s.agentEndVersion)
+	require.NoError(s.T(), err)
+
+	require.Eventually(s.T(), agentStatus("online", *s), 5*time.Minute, 5*time.Second, "Agent status is not online")
+
+	newVersion, err := tools.GetAgentVersion(s.requirementsInfo.KibanaClient)
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), s.agentEndVersion, newVersion)
+}
+
+func (s *UpgradeElasticAgent) TearDownTest() {
+	s.T().Log("Un-enrolling elastic agent")
+	assert.NoError(s.T(), tools.UnEnrollAgent(s.requirementsInfo.KibanaClient))
+
+	s.T().Log("Uninstalling elastic agent")
+	output, err := tools.UninstallAgent(s.agentFixture)
+	if err != nil {
+		s.T().Log(string(output))
+	}
+	require.NoError(s.T(), err)
+}
+
+func TestElasticAgentUpgrade(t *testing.T) {
+	currentVersion := version.GetDefaultVersion()
+	previousVersion, err := getPreviousMinorVersion(currentVersion)
+	require.NoError(t, err)
+
+	info := define.Require(t, define.Requirements{
```

Comment:
This has to be the first function call in the test. The `define.Require` will always use the version that is from `define.Version()` in the case that the stack version is not defined. So you can just do `Stack: &define.Stack{}`, that will ensure its the same version of `define.Version()`.

Reply:
> As this stands this should fail `mage integration:check` and `mage check`.

Yes, it does.  👌 

```
$ mage integration:check
>> check: Checking for define.Require in integration tests
Error: test TestElasticAgentUpgrade first statement must be a function call to define.Require
```